src/api/rate_limiter.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import requests
+import uuid
+
+from datetime import datetime
+
+
+class RequestEntry(object) :
+    """
+    Class for request object to put in the queue for the rate limiter
+    """
+    def __init__(self,url,cookies,key,method) -> None:
+        self.url = url
+        self.cookies = cookies
+        self.key = key
+        self.method = method
+
+class RateLimiter() :
+    """
+    Class that takes care of sending the requests at a maximum rate (25/sec) 
+        and giving back the result to the calling functions when it is received (async probably)
+    """
+    def __init__(self) :
+        self.requests = {}
+        self.queue = asyncio.Queue()
+        asyncio.create_task(self.handle_requests())
+
+
+    async def handle_requests(self) :
+        print(f"Starting rate_limiter task...")
+        while True :
+            #take a new request from the queue
+            request = await self.queue.get()
+            print(f"[{datetime.now().strftime('%H:%M:%S')}] Treating item in queue : {request.key} -> {request.url} + {request.cookies} ")
+
+            # wait 50ms for rate limitation purpose ;)
+            await asyncio.sleep(0.050)
```

Comment:
It works but might be overkill
The request already takes some time here, so we may wait too long
What about storing the last request time and add instead of the sleep something like
```py
while datetime.now() - lastRequestTime < 0.050:
    continue
```

Reply:
True

Reply:
actually that piece of code is really bad habits, because it not only slows the code it also slows the machine: while a `time.sleep` give the processor back to the kernel this "active waiting" doesn't! 
I would prefer 
```py
delay = datetime.now() - lastRequestTime - 0.050
if delay > 0.001 :
  time.sleep(delay)
  ```

Reply:
I agree about the time.sleep() but the point of using asyncio.sleep (to be confirmed but that's what I understood) is that it gives this time not to the processor but to other asynchronous tasks that might be waiting somewhere else in the program. So (to me) time.sleep would be fully blocking the program, whereas asyncio.sleep is allowing the rest of the program to still run while this particular task waits.

Reply:
Actually my bad, my point wasn't the `time.sleep` or `asyncio.sleep` it was of not using `while... continue` 
```py
delay = datetime.now() - lastRequestTime - 0.050
if delay > 0.001 :
  asyncio.sleep(delay)
  ```
  is fine to me

Reply:
fixed :+1: